cmd/xl-storage.go

```diff
@@ -644,6 +652,12 @@ func (s *xlStorage) NSScanner(ctx context.Context, cache dataUsageCache, updates
 				}
 			}
 		}
+		if objDeleted {
+			// we return errSkipFile to signal this function's
+			// callers to skip this object's contribution towards
+			// usage.
+			return sizeSummary{}, errSkipFile
```

Comment:
I think we should not return an error here; this looks like a clarity simplificiation but this can cause a performance regression; basically all expired objects may go to a healing check performed by the caller using abondonedChildren mechanism (objects found in another disk in the last scan but not found in the current disk, we execute healing of those objects)

/cc @klauspost

Reply:
@vadmeste Your concern is valid. @klauspost I need a way to signal from within the `getSize` callback function that this `item` doesn't contribute toward data usage. I have added a commit on top of my PR to address @vadmeste 's concern. Let me know if this is acceptable.